changelogs/2.10.0.md

```diff
@@ -0,0 +1,933 @@
+
+[release_policy]: https://www.tarantool.io/en/doc/latest/dev_guide/release_management/#release-policy
+[issues]: https://github.com/tarantool/tarantool/issues
+
+## Compatibility
+
+Tarantool 2.x is backward compatible with Tarantool 1.10.x in the binary data
+layout, client-server protocol, and replication protocol.
+
+Please [upgrade][upgrade] using the `box.schema.upgrade()` procedure to unlock
+all the new features of the 2.x series.
+
+Following is a list of changes which break backward compatibility:
+* `fiber.wakeup()` in Lua and `fiber_wakeup()` in C became NOP
+on the currently running fiber.
+* `timeout()` method of `net.box` connection was dropped.
+* net.box console support was dropped.
+* lua: return value type for all time64 functions changed from
+`uint64_t` to `int64_t`.
+
+[upgrade]: https://www.tarantool.io/en/doc/latest/book/admin/upgrades/
+
+## Functionality added or changed
+
+### Core
+
+* Field type `UUID` is now a part of field type `SCALAR` (gh-6042).
+* Field type `UUID` is now available in SQL, and a new `UUID` can be generated
+  using the new SQL built-in `uuid()` function (gh-5886).
+
+* **[Breaking change]** `timeout()` method of `net.box` connection, which was
+  marked deprecated more than four years ago (in 1.7.4), was dropped, because
+  it negatively affected performance of hot `net.box` methods, like `call()`
+  and `select()`, in case those were called without specifying a timeout
+  (gh-6242).
+
+* Improved `net.box` performance by up to 70% by rewriting hot code paths in C
+  (gh-6241).
+* Introduced compact tuples that allow saving 4 bytes per tuple in case of
+  small userdata (gh-5385).
+* Streams and interactive transactions over streams are implemented
+  in iproto. A stream is associated with its ID, which is unique within
+  one connection. All requests with the same not zero stream ID belong to
+  the same stream. All requests in the stream are processed synchronously.
+  The execution of the next request will not start until the previous one is
+  completed. If a request has zero stream ID, it does not belong to a stream
+  and is processed in the old way.
+
+  In `net.box`, a stream is an object above connection that has the same
+  methods but allows to execute requests sequentially. ID is generated
+  automatically on the client side. If users write their own connector and
+  want to use streams, they must transmit stream_id over iproto protocol.
+  The main purpose of streams is transactions via iproto.
+  Each stream can start its own transaction, allowing multiplexing several
+  transactions over one connection. There are multiple ways to begin,
+  commit and rollback a transaction: using appropriate stream methods, using
+  `call` or `eval` methods or using `execute` method with SQL transaction
+  syntax. Users can mix these methods, for example, start a transaction using
+  `stream:begin()`, and commit a transaction using `stream:call('box.commit')`
+  or `stream:execute('COMMIT')`.
```

Comment:
issue label is missing

Reply:
https://github.com/tarantool/tarantool/issues/5860